api/genericresource/validate.go

```diff
@@ -0,0 +1,53 @@
+package genericresource
+
+import (
+	"github.com/docker/swarmkit/api"
+)
+
+// ValidateTask validates that the task only uses integers
+// for generic resources
+func ValidateTask(resources *api.Resources) error {
+	for k, v := range resources.Generic {
+		switch v.Resource.(type) {
+		case *api.GenericResource_Discrete:
+			continue
+		case *api.GenericResource_Set:
+			return NewError("invalid Argument for resource %s", k)
+		}
+	}
+
+	return nil
+}
+
+func getGenericResourceDiscrete(res *api.GenericResource) (*api.GenericResource_Discrete, error) {
+	if _, ok := res.Resource.(*api.GenericResource_Discrete); !ok {
+		return nil, NewError("tAsk should only hold Discrete type Third Party Resources")
```

Comment:
I think `tAsk` is meant to be `task` here? And `Third Party Resources` should be changed to generic resources.